src/grid/grid.directive.ts

```diff
@@ -0,0 +1,57 @@
+import { Directive, HostBinding, Input, OnInit } from "@angular/core";
+
+@Directive({
+	selector: "[ibmGrid]"
+})
+export class GridDirective {
+	@HostBinding("class.bx--grid") baseClass = true;
+}
+
+@Directive({
+	selector: "[ibmRow]"
+})
+export class RowDirective {
+	@HostBinding("class.bx--row") baseClass = true;
+}
+
+@Directive({
+	selector: "[ibmCol]"
+})
+export class ColumnDirective implements OnInit {
+	@Input()
+	class = "";
+
+	@Input()
+	columnNumbers = {};
+
+	@Input()
+	offsets = {};
+
+	@HostBinding("class")
+	columnClasses: string;
+
+	ngOnInit() {
+		const classNames = [];
+		try {
```

Comment:
Just want to double check why this is in a try/catch ...

Reply:
Just in case `columnNumbers` or `offsets` are not in the right format I was thinking it was good to catch the error and log it.

Reply:
Yeah that makes sense :+1: probably want to add a clarifying message to the log then ... I'll do up a suggestion